src/main/java/com/faforever/client/os/FileOpeningHandler.java

```diff
@@ -49,11 +49,26 @@ private void runReplay(Path filePath) {
     }
   }
 
+  private void hostReplay(Path filePath) {
+    try {
+      replayService.hostFromReplayFile(filePath);
+    } catch (CompressorException | IOException e) {
+      notificationService.addImmediateErrorNotification(e, "replay.couldNotParse");
+    }
+  }
+
   @Override
   public void run(ApplicationArguments args) {
     String[] sourceArgs = args.getSourceArgs();
     if (sourceArgs.length > 0) {
       runReplay(Path.of(sourceArgs[0]));
     }
   }
+
+  public void host(ApplicationArguments args) {
+    String[] sourceArgs = args.getSourceArgs();
+    if (sourceArgs.length > 0) {
+      hostReplay(Path.of(sourceArgs[0]));
+    }
+  }
```

Comment:
What is the purpose of this? If we open a replay from the file system it is more likely they want to watch it not host a game like it.